src/main/java/igrad/logic/parser/module/ModulePreclusionParser.java

```diff
@@ -0,0 +1,63 @@
+package igrad.logic.parser.module;
+
+import java.util.Arrays;
+
+/**
+ * Parser for modules' preclusion data from NUSMods.
+ */
+public class ModulePreclusionParser {
+
+    private String preclusionString;
+    private String[] preclusionModules;
+
+    public ModulePreclusionParser(String preclusionString) {
+        this.preclusionString = preclusionString;
+        this.preclusionModules = splitPreclusionModules(this.preclusionString);
+    }
+
+    /**
+     * Returns a String array of preclusion module codes in String form from data.
+     */
+    public String[] splitPreclusionModules(String preclusionModulesString) {
+        String[] preclusionModules = preclusionModulesString.split(" ");
+        preclusionModules = Arrays.stream(preclusionModules).filter(x -> isModule(x)).toArray(String[]::new);
+        preclusionModules = Arrays.stream(preclusionModules)
+                .map(x -> removeModulePunctuation(x)).toArray(String[]::new);
+
+        return preclusionModules;
+    }
+
+    /**
+     * Verifies if String given is a valid module code.
+     */
+    private boolean isModule(String module) {
```

Comment:
there's a duplication in logic here, you could actually use `ModuleCode.isValidModuleCode(String test)`